### ISSUE TITLE
warning: field 'retrieved_headers_' will be initialized after field 'added_headers'

### ISSUE THREAD

**Issue by HighCommander4** [cpp-netlib/cpp-netlib#814]
When compiling a program that includes cpp-netlib headers with clang 4.0 with `-Wall`, I get warnings like this:

```
In file included from /home/nr/dev/dist/cpp-netlib/include/boost/network/protocol/http/client.hpp:11:
In file included from /home/nr/dev/dist/cpp-netlib/include/boost/network/protocol/http/request.hpp:10:
In file included from /home/nr/dev/dist/cpp-netlib/include/boost/network/protocol/http/impl/request.hpp:26:
/home/nr/dev/dist/cpp-netlib/include/boost/network/protocol/http/message/async_message.hpp:58:9: warning: field 'retrieved_headers_' will be initialized after field 'added_headers'
      [-Wreorder]
        retrieved_headers_(other.retrieved_headers_),
        ^
```

Using cpp-netlib 0.13.0-rc1.

**Comment by deanberris** [comment]
This should be fixed in the latest in 0.13-release -- can you confirm? If so the next RC should have this (which I intend to get to sometime soon).

**Comment by HighCommander4** [comment]
Confirmed fixed on the 0.13-release branch, thanks.
